be/src/olap/schema_change.cpp

```diff
@@ -1649,8 +1650,8 @@ Status VSchemaChangeWithSorting::_internal_sorting(
     MultiBlockMerger merger(new_tablet);
 
     std::unique_ptr<RowsetWriter> rowset_writer;
-    RETURN_IF_ERROR(
-            new_tablet->create_rowset_writer(version, VISIBLE, segments_overlap, &rowset_writer));
+    RETURN_IF_ERROR(new_tablet->create_rowset_writer(version, VISIBLE, segments_overlap,
+                                                     &new_tablet->tablet_schema(), &rowset_writer));
```

Comment:
If create rowset writer always bind the tablet's tablet_schema, then its no need as an input parameter. We could get the tablet schema during create_rowset_writer inside.

Reply:
it is not only use for new_tablet. Such as PushHandler::_convert_v2, Compaction::construct_output_rowset_writer. They should set newer schema from outside.